pkg/kubelet/network/plugins.go

```diff
@@ -47,6 +48,22 @@ type NetworkPlugin interface {
 
 	// TearDownPod is the method called before a pod's infra container will be deleted
 	TearDownPod(namespace string, name string, podInfraContainerID kubeletTypes.DockerID) error
+
+	// Status is the method called to obtain the ipv4 and pv6 addresses of the container
+	Status(namespace string, name string, podInfraContainerID kubeletTypes.DockerID) (*NetworkStatus, error)
+}
+
+type InterfaceInfo struct {
+	Iface   net.Interface `json:"iFace" description:"A network interface on the pod"`
+	IPAddrs []net.IP      `json:"ipAddrs" description:"List of IP Addresses associated with the given interface"`
+}
+
+// Stores the network status of a pod (currently just the network addresses and interfaces)
+type NetworkStatus struct {
```

Comment:
as a JSON struct this needs a Kind and APIVersion field, just like api.TypeMeta.  In fact, just embed an api.TypeMeta.  See plugin/pkg/scheduler/api/types.go for precedent.